db/infinispan/src/main/java/org/commonjava/indy/infinispan/data/InfinispanStoreDataMigrationAction.java

```diff
@@ -85,7 +85,7 @@ public boolean migrate() throws IndyLifecycleException
     {
         String clsName = storeDataManager.getClass().getName();
 
-        if ( !( storeDataManager instanceof InfinispanStoreDataManager ) )
+        if ( !( storeDataManager instanceof InfinispanStoreDataManager && ((InfinispanStoreDataManager) storeDataManager).isStandaloneMode() ) )
```

Comment:
This class casting is really overused in many places in this pr. So if it will happen so many times, I'd recommend to add the method isStandaloneMode() (Or what anything else it will be) to interface StoreDataManager to avoid the casting. We can add it with a default implementation to return false or something else if it will change.